core/os/thread.cpp

```diff
@@ -64,3 +103,14 @@ Error Thread::set_name(const String &p_name) {
 
 	return ERR_UNAVAILABLE;
 }
+
+Thread::~Thread() {
+	if (id != 0) {
+#ifdef DEBUG_ENABLED
+		WARN_PRINT("A Thread object has been destroyed without wait_to_finish() having been called on it. Please do so to ensure correct cleanup of the thread.");
```

Comment:
Is this still really needed? Documentation states resources are freed once the thread exits: https://en.cppreference.com/w/cpp/thread/thread/detach

Reply:
My concern is that the code running on the thread, if left to run widly, may access something that is already released on in a wrong state.
Maybe in practice that's unlikely, but I thought it was a good idea to just warn against fire-and-forget, so joining the thread is the _de facto_ standard practice, which can help catching bugs.